app/mm-redux/actions/channel_categories.ts

```diff
@@ -1,18 +1,534 @@
+import {General} from '@mm-redux/constants';
+import {CategoryTypes} from '@mm-redux/constants/channel_categories';
+import {getAllCategoriesByIds, getCategory, getCategoryIdsForTeam, getCategoryInTeamByType, getCategoryInTeamWithChannel} from '@mm-redux/selectors/entities/channel_categories';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
+import {getUser as selectUser, getUserIdsInChannels} from '@mm-redux/selectors/entities/users';
+import {ActionFunc, batchActions, DispatchFunc, GetStateFunc} from '@mm-redux/types/actions';
+import {CategorySorting, ChannelCategory, OrderedChannelCategories} from '@mm-redux/types/channel_categories';
+import {Channel} from '@mm-redux/types/channels';
+import {UserProfile} from '@mm-redux/types/users';
+import {$ID, RelationOneToMany} from '@mm-redux/types/utilities';
+import {insertMultipleWithoutDuplicates, insertWithoutDuplicates, removeItem} from '@mm-redux/utils/array_utils';
+import {getUserIdFromChannelName} from '@mm-redux/utils/channel_utils';
+
+import {favoriteChannel, getChannelMembersByIds, unfavoriteChannel} from './channels';
+import {logError} from './errors';
+import {forceLogoutIfNecessary} from './helpers';
 
 export function expandCategory(categoryId: string) {
-    return {
-        type: ChannelCategoryTypes.CATEGORY_EXPANDED,
-        data: categoryId,
-    };
+    return setCategoryCollapsed(categoryId, false);
 }
 
 export function collapseCategory(categoryId: string) {
+    return setCategoryCollapsed(categoryId, true);
+}
+
+export function setCategoryCollapsed(categoryId: string, collapsed: boolean) {
+    return patchCategory(categoryId, {
+        collapsed,
+    });
+}
+
+export function setCategorySorting(categoryId: string, sorting: CategorySorting) {
+    return patchCategory(categoryId, {
+        sorting,
+    });
+}
+
+export function patchCategory(categoryId: string, patch: Partial<ChannelCategory>): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
+        const currentUserId = getCurrentUserId(state);
+
+        const category = getCategory(state, categoryId);
+        const patchedCategory = {
+            ...category,
+            ...patch,
+        };
+
+        // Optimistic Client
+        dispatch({
+            type: ChannelCategoryTypes.RECEIVED_CATEGORY,
+            data: patchedCategory,
+        });
+
+        try {
+            await Client4.updateChannelCategory(currentUserId, category.team_id, patchedCategory);
+        } catch (error) {
+            dispatch({
+                type: ChannelCategoryTypes.RECEIVED_CATEGORY,
+                data: category,
+            });
+
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        return {data: patchedCategory};
+    };
+}
+
+export function setCategoryMuted(categoryId: string, muted: boolean) {
```

Comment:
Why this cannot follow the same approach as expand or collapse category (just call patch category with the new information)?

Reply:
I think this one I pulled from the webapp where it doesn't need the optimistic dispatch that patch provides.